### ISSUE TITLE
Resultados dos Experimentos - Otimizadores

### ISSUE THREAD

**Issue by Erickslb** [Erickslb/deep-learning-fgv-2026#1]
Poste seus resultados aqui. Lembre de colocar:

- Seu nome e sobrenome;
- Pergunta/hipótese que quis responder — **escreva antes de rodar**, não depois;
- Quais otimizadores e learning rates você comparou;
- Imagem dos gráficos gerados;
- Breve comentário explicando as observações.

**Comment by nicolasspaniol** [comment]
**Nome**: Nícolas Spaniol
**Hipótese testada**: Adam aprende melhor que SGD, independente da escala da LR
**Otimizadores/LRs comparados**:
```python
otimizadores = {
    "SGD (lr=0.01)": lambda params: torch.optim.SGD(params, lr=0.01),
    "Adam (lr=0.01)": lambda params: torch.optim.Adam(params, lr=0.01),
    "Adam (lr=0.3)": lambda params: torch.optim.Adam(params, lr=0.3),
    "SGD (lr=0.3)": lambda params: torch.optim.SGD(params, lr=0.3),
}
```
**Resultados**: <img width="1991" height="490" alt="Image" src="https://github.com/user-attachments/assets/d50ffa54-0079-4c7f-98bc-1d928e6a994e" />

Fiquei surpreso que, nessa run, o Adam não conseguiu aprender com LR=0.3. Como o método tem _learning rate_ adaptativa, achei que sua performance não seria tão dependente do valor definido inicialmente. Com LR=0.01, o Adam de fato se saiu melhor, mas não por muito.

**Comment by alexoliveiraFGV24** [comment]
**Nome**: Alex Júnio Maia de Oliveira
**Hipótese**: Sabemos que o Adam e o RMSProp utilizam learning rates adaptativos, calculados com base no histórico dos gradientes da função objetivo. Qual converge mais rápido com um mesmo learning rate inicial?
**Otimizadores**: 
```bash
otimizadores = {
    "RMSProp (lr=0.001)": lambda params: torch.optim.RMSprop(params, lr=0.001),
    "RMSProp (lr=0.01)": lambda params: torch.optim.RMSprop(params, lr=0.01),
    "Adam (lr=0.01)": lambda params: torch.optim.Adam(params, lr=0.01),
    "Adam (lr=0.001)": lambda params: torch.optim.Adam(params, lr=0.001),
}
````

<img width="1716" height="422" alt="Image" src="https://github.com/user-attachments/assets/cbc4b3af-c54b-4f00-a8b4-6adfca77eea4" />

**Comentários**: O meu pensamento inicial foi de que o Adam teria um performance melhor, e os gráficos confirmaram a hipótese. Interessante de que, ao fazer alguns testes, quando colocamos um learning rate alto para RMSProp (por exemplo 0.1) o erro explode.
